### PR TITLE
Setup Associations

### PR DESCRIPTION
**Pull request by aamir-asaram**
**In this PR, I have implemented the following:**
- Created `Classroom`, `Book` and `Rental` classes
- Created a has-many relationship between the `Classroom` and `Student` classes
- Created `has-many` and `many-to-many` relationships between the `Person` and `Book` classes with the help of the `Rental` helper class

### CODE REVIEW

File: student.rb [hunk 1]
```diff
@@ -4,6 +4,7 @@ class Student < Person
   def initialize(age, classroom, parents_permission: true, name: 'Unknown')
     super(age, parents_permission: parents_permission, name: name)
     @classroom = classroom
+    classroom.add_student(self)
   end
 
   def play_hooky
```
Comment (ShadyShawkat) [inline student.rb]: - [ ] Kindly add a method that serves as the 'belongs-to' side (a student belongs to a classroom). Like so:
```ruby
  def classroom=(classroom)
      @classroom = classroom
      classroom.students.push(self) unless classroom.students.include?(self)
  end
```

**Review by ShadyShawkat — CHANGES_REQUESTED** [review]
## Hi @aamir-asaram, 

Good job so far! 👍🏻 
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight
- Correct use of GitFlow ✔️ 
- Descriptive commit messages ✔️ 
- Professional looking pull request ✔️ 

Image

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: student.rb (hunk 1 above)
Comment (aamir-asaram) [reply student.rb]: I completely missed that, thanks for pointing it out! @ShadyShawkat :smile:

**Review by codepantha — APPROVED** [review]
# STATUS: APPROVED 🟢 ✔️ 

## Hi @aamir-asaram 👋, 

Great job implementing the required changes 👏 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights 🎯 
✔️ Descriptive PR
✔️ Good commit history
✔️ Created classes and the associations between them 📌 
✔️ Linters are passing 🟢 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**